db/migrations/20180102183100_add_stack_to_buildpack_table.rb

```diff
@@ -0,0 +1,30 @@
+require 'yaml'
+
+def default_stack
+  stacks_yml_path = ENV.fetch('STACKS_YML', nil)
+  YAML.safe_load(File.read(stacks_yml_path))['default'] if stacks_yml_path && File.exist?(stacks_yml_path)
+end
+
+Sequel.migration do
+  up do
+    alter_table(:buildpacks) do
+      add_column :stack, String, size: 255, null: true
+      drop_index :name, unique: true
+      add_index [:name, :stack], unique: true, name: :unique_name_and_stack
+    end
+
+    self['UPDATE buildpacks SET stack = ?', default_stack || 'unknown'].update
```

Comment:
We would prefer to not use raw SQL when possible. Is there a way to write this using the Sequel migration DSL?

Reply:
We'll give it a shot and reach out if we can't.